### DISCUSSION TITLE
Python error on Verce;

### DISCUSSION THREAD

**Discussion by SaraswatAnushka** [vercel/community#6336] [Help]
### Summary

I am trying to host my Flask app on vercel but it keeps giving this error: Error: Unable to find any supported Python versions.

### Steps to Reproduce

This is the error 
<img width="1440" alt="Screenshot 2024-03-30 at 7 39 22 AM" src="https://github.com/vercel/community/assets/96346577/67a4e773-ea1a-4f7b-92dd-c10a810d4928">
and this is my vercel.json file
{
    "rewrites": [
      { "source": "/(.*)", "destination": "/api/index.py" }
    ]
}

package.json file
{
    "engines": {
      "node": "18.x"
    }
  }
  Please help

**Comment by gsnoopy** [comment]
To run flask in vercel run the command: _**pip install pipenv**_ in your project
now run the command:  _**pipenv lock**_ this command creates a Pipfile and Pipfile.lock in your project
finalize with command _**pipenv install flask**_ and pipenv goes update your pipfile with flask.

In Pipfile your config will be:

[[source]]
url = "https://pypi.org/simple"
verify_ssl = true
name = "pypi"
 
[packages]
flask = "*"
 
[requires]
python_version = "3.9"

and you need a requirements.txt with:

`Flask==2.2.2`

Read:
- [Vercel Document](https://vercel.com/docs/functions/runtimes/python)
- [Pipenv tutorial](https://dev.to/davidshare/setting-up-a-python-development-environment-with-pipenv-3lfj)

**Comment by amyegan** [comment]
Thanks for reporting this. The engineering team is working on a bugfix now. But you can find a workaround posted in https://github.com/vercel/community/discussions/6287#discussioncomment-8932365
